extensions/azure/data-plane/data-factory/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/DataPlaneAzureDataFactoryExtension.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.azure.dataplane.azuredatafactory;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.datafactory.DataFactoryManager;
+import com.azure.security.keyvault.secrets.SecretClientBuilder;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.TransferService;
+import org.eclipse.dataspaceconnector.dataplane.spi.registry.TransferServiceRegistry;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.system.SettingResolver;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * Registers a {@link TransferService} for performing data transfers with Azure Data Factory.
+ */
+public class DataPlaneAzureDataFactoryExtension implements ServiceExtension {
+
+    @EdcSetting
+    private static final String KEY_VAULT_LINKED_SERVICE_NAME = "edc.data.factory.key.vault.linked.service.name";
```

Comment:
nit: is there a hierarchy to this? otherwise I'd prefer `...linked-service-name`

Reply:
Not sure this use of dashes would play out well with how we resolve system properties/env variables

Reply:
underscores then

Reply:
That would make it impossible to set it via env variables. Those underscores would be replaced with periods...

Reply:
then lets just join them as "linkedservicename". not ideal, but (to my mind) better than the dots.